src/Chartjs/assets/src/chart_v4_controller.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import AbstractChartController from './abstract_controller';
+import Chart from 'chart.js/auto';
```

Comment:
I suspect that this is caused by something "odd" with our setup (I believe it's because Chart.js 4 is ESM-only) but to fix the test:

```suggestion
import Chart from 'chart.js/auto/auto.cjs';
```

But I'm not sure that's the proper solution... and it seems like, from a user's perspective, `chart.js/auto` DOES work, and we shouldn't need 2 controllers to solve this problem. Again, I've got the feeling that we're doing something silly...

Reply:
Well that indeed seemed to fix the problem, and the tests in the PR are still failing because of reason mentioned in the todo-list.

Yeah I got that hunch too, seems like we are only satisfying the test-suite for some reason. But that part goes beyond my current js (test-suite) knowledge unfortunately...

Reply:
I looked at something like [resolve](https://www.npmjs.com/package/resolve), to dynamically check for the module existence so we could still have a single controller. But couldn't get that to work 😞

Reply:
Btw, I think that:

A) Our code already works with V4 and we just need to allow it in `package.json`. I think you and/or other users have already verified this.
B) The actual test failing problem is a problem with our tests - specifically, I think jest is behaving poorly because it doesn't it doesn't play well with modules.

Possible solution is to move from jest to https://vitest.dev/